api/envoy/extensions/load_balancing_policies/least_request/v3/least_request.proto

```diff
@@ -60,9 +67,7 @@ message LeastRequest {
   // Configuration for local zone aware load balancing or locality weighted load balancing.
   common.v3.LocalityLbConfig locality_lb_config = 4;
 
-  // [#not-implemented-hide:]
-  // Configuration for performing full scan on the list of hosts.
-  // If this configuration is set, when selecting the host a full scan on the list hosts will be
-  // used to select the one with least requests instead of using random choices.
-  google.protobuf.BoolValue enable_full_scan = 5;
+  // Method for selecting the host set from which to return the host with the fewest active requests.
+  // Defaults to POWER_OF_N_CHOICES, where N is determined by `choice_count` (2 by default).
+  SelectionMethod selection_method = 5 [(validate.rules).enum = {defined_only: true}];
```

Comment:
Are there any issues with modifying the meaning of field 5 when it was never included in an Envoy release (AFAIK)?

Reply:
Yes, this is a problem.  The xDS protos are not used exclusively for Envoy, and they do get picked up by other data planes (e.g., gRPC) in between Envoy releases.

Please leave the existing field as-is, although you can mark it as deprecated if you want. The new field should be added with a new number, and you can document that the old field is ignored if the new field is set.